test/ech_test.c

```diff
@@ -34,6 +562,310 @@ const OPTIONS *test_get_options(void)
+    }
+    if (!TEST_int_eq(OSSL_ECHSTORE_write_pem(es, OSSL_ECHSTORE_ALL, out), 1)) {
+        TEST_info("OSSL_ECSTORE_write_pem unexpected fail");
+        goto end;
+    }
+    now = time(0);
+    if (!TEST_int_eq(OSSL_ECHSTORE_flush_keys(es, now), 1)) {
+        TEST_info("OSSL_ECSTORE_flush_keys unexpected fail");
+        goto end;
+    }
+    if (!TEST_int_eq(OSSL_ECHSTORE_num_keys(es, &keysaftr), 1)) {
+        TEST_info("OSSL_ECSTORE_num_keys unexpected fail");
+        goto end;
+    }
+    if (!TEST_int_eq(keysaftr, 0)) {
+        TEST_info("OSSL_ECSTORE_flush_keys unexpected non-zero");
+        goto end;
+    }
+    rv = 1;
+end:
+    OSSL_ECHSTORE_free(es);
+    BIO_free_all(in);
+    BIO_free_all(out);
+    return rv;
+}
+
+/* make a bunch of calls with bad, mostly NULL, arguments */
+static int ech_store_null_calls(void)
+{
+    int rv = 0, count = 0;
+    OSSL_ECHSTORE *es = OSSL_ECHSTORE_new(NULL, NULL);
+    OSSL_HPKE_SUITE hpke_suite = OSSL_HPKE_SUITE_DEFAULT;
+    BIO *inout = BIO_new(BIO_s_mem());
+    OSSL_ECH_INFO *info = NULL;
+    EVP_PKEY *priv = EVP_PKEY_new();
+
+    OSSL_ECHSTORE_free(NULL);
+    if (!TEST_int_eq(OSSL_ECHSTORE_new_config(NULL, OSSL_ECH_CURRENT_VERSION,
+                                              0, "example.com", hpke_suite),
+                     0)) {
+        TEST_info("OSSL_ECHSTORE_new_config unexpected non-zero");
+        goto end;
+    }
+    if (!TEST_int_eq(OSSL_ECHSTORE_new_config(es, OSSL_ECH_CURRENT_VERSION,
+                                              0, NULL, hpke_suite),
+                     0)) {
+        TEST_info("OSSL_ECHSTORE_new_config unexpected non-zero");
+        goto end;
+    }
+    hpke_suite.kdf_id = 0xAAAA; /* a bad value */
+    if (!TEST_int_eq(OSSL_ECHSTORE_new_config(es, OSSL_ECH_CURRENT_VERSION,
+                                              0, "example.com", hpke_suite),
+                     0)) {
+        TEST_info("OSSL_ECHSTORE_new_config unexpected non-zero");
+        goto end;
+    }
+    if (!TEST_int_eq(OSSL_ECHSTORE_write_pem(NULL, 0, inout), 0)) {
+        TEST_info("OSSL_ECHSTORE_write_pem unexpected non-zero");
+        goto end;
+    }
+    if (!TEST_int_eq(OSSL_ECHSTORE_write_pem(es, 0, NULL), 0)) {
+        TEST_info("OSSL_ECHSTORE_write_pem unexpected non-zero");
+        goto end;
+    }
+    if (!TEST_int_eq(OSSL_ECHSTORE_write_pem(es, 100, inout), 0)) {
+        TEST_info("OSSL_ECHSTORE_write_pem unexpected non-zero");
+        goto end;
+    }
+    if (!TEST_int_eq(OSSL_ECHSTORE_read_echconfiglist(NULL, inout), 0)) {
+        TEST_info("OSSL_ECHSTORE_read_echconfiglist unexpected non-zero");
+        goto end;
+    }
+    if (!TEST_int_eq(OSSL_ECHSTORE_read_echconfiglist(es, NULL), 0)) {
+        TEST_info("OSSL_ECHSTORE_read_echconfiglist unexpected non-zero");
+        goto end;
+    }
+    if (!TEST_int_eq(OSSL_ECHSTORE_get1_info(NULL, &info, &count), 0)) {
+        TEST_info("OSSL_ECHSTORE_get1_info unexpected non-zero");
```

Comment:
Strictly speaking, `info` is now allocated and now needs to be freed. Similarly with the next few lines.

Reply:
Ok - added. I guess one could argue that if you hit that leak though, then there's breakage to fix anyway, so maybe better to see the leak via the test?  IOW, I'd be a very tiny bit more in favour  of taking out the `OSSL_ECH_INFO_free()` call in the 'null_calls' test, but am ok including it if preferred. (It's included for now.)